rust/connlib/libs/tunnel/src/tun_darwin.rs

```diff
@@ -115,54 +95,61 @@ impl IfaceDevice {
         }
     }
 
-    pub async fn new(name: &str) -> Result<Self> {
-        let idx = parse_utun_name(name)?;
-
-        let fd = match unsafe { socket(PF_SYSTEM, SOCK_DGRAM, SYSPROTO_CONTROL) } {
-            -1 => return Err(get_last_error()),
-            fd => fd,
-        };
-
+    // On Apple platforms, we must use a NetworkExtension for reading and writing
+    // packets if we want to be allowed in the iOS and macOS App Stores. This has the
+    // unfortunate side effect that we're not allowed to create or destroy the tunnel
+    // interface ourselves. The file descriptor should already be opened by the NetworkExtension for us
+    // by this point. So instead, we iterate through the 1024 possible file descriptors
+    // looking for the one corresponding to the utun interface we have access to read and
+    // write from.
+    //
+    // Inspired heavily by WireGuard's implementation for Apple:
+    // https://github.com/WireGuard/wireguard-apple/blob/master/Sources/WireGuardKit/WireGuardAdapter.swift
+    pub async fn new(_: Option<&str>) -> Result<Self> {
```

Comment:
What is the purpose of the `Option<&str>` argument?

Reply:
On Linux platforms, we _do_ want to create the interface, so on those platforms with pass in a string argument to use for the interface name. The Option allows up to have a single API to adhere to for Unix platforms for the remaining WireGuard stack.

Reply:
See https://github.com/firezone/firezone/blob/jamilbk-feat/apple-tunnel-stack/rust/connlib/libs/tunnel/src/device_channel_unix.rs#L64